src/window.rs

```diff
@@ -157,13 +151,15 @@ pub struct WindowAttributes {
     pub resize_increments: Option<Size>,
     pub content_protected: bool,
     pub window_level: WindowLevel,
-    pub parent_window: Option<RawWindowHandle>,
+    #[cfg(feature = "rwh_06")]
+    pub parent_window: Option<rwh_06::WindowHandle<'a>>,
```

Comment:
While I don't have time to review this yet, I want to comment on this particular matter. We basically need to account for the use case where you send a `WindowBuilder` across the threads.

However our current builder also can't be `Send`. Maybe we should somehow separate the `!Send` part out of the structure? This will also move the lifetime to some other struct.

We need to avoid the lifetime on the builder, but we also need to avoid `rwhnd` and `!Send` all together. There was an issue about that somewhere, but I don't remember where...

Reply:
On macOS (and iOS), you could create a `Send`/`Sync` handle with something like:
```rust
struct OwnedWindowHandle {
    view: MainThreadBound<Id<NSView>>,
}

impl OwnedWindowHandle {
    fn from_handle(handle: rwh_06::WindowHandle<'_>) -> Self {
        let view = match handle.raw_window_handle() {
            // SAFETY: The safety contract of WindowHandle<'_> ensures the pointer is valid.
            //
            // `NSView` may only be used on the main thread, but that is ensured below
            // by putting it in a MainThreadBound.
            RawWindowHandle::AppKit(handle) => unsafe { Id::retain(handle.ns_view.as_ptr().cast()) }.unwrap();
            handle => panic!("Invalid window handle {handle:?} on macOS"),
        };
        // SAFETY: NSView is safe to Drop on the main thread.
        let mtm = unsafe { MainThreadMarker::new_unchecked() };
        let view = MainThreadBound::new(view, mtm);
        Self { view }
    }
}
```

Reply:
No, I mean, could we simply split the `WindowBuilder` in a way that once you pass the `with_parent` it becomes not `Send`, but by default it's `Send`? I don't want a send/sync handle, I just want most parts of the builder being send and have a way to make it into the non-send version?

Reply:
Like
```
let builder = WindowBuilder::new().with_title("Foo"); // This you can send.
let not_send = builder.with_parent(Some(parent)); // This you can't send anymore.
```

Reply:
Well, it simply means that you mark rwhnd as safe to send + sync, while the real one is not. It's true that Wayland one is send + sync, but from a users point of view what I suggest is zero changes. it's just our internals will have a bit more code, but nothing really bad.

Reply:
And RawWindowHandle is just one example, you can literally add anything, like raw X11 stuff to build a window, some random pointers, etc. You can solve the issue for one type of things, but not for other.

Also, the reason I suggested, is because there's a desire to send window builders between threads, but finish building on other threads.

Reply:
By the complication in API surface, I mean that your suggestion means we must introduce a new `ThreadUnsafeWindowBuilder`, which will be confusing for users for little benefit in my opinion.

> raw X11 stuff to build a window, some random pointers

Seems like we should make those thread safe as well, then.

Reply:
Maybe you haven't done that, but `WindowBuilder<T = ()>` and then `with_parent(self) -> WindowBuilder<T = NotSendUdata>` will work just as fine.

Reply:
I can do this all myself in a separate patch though.

Reply:
That'd work, but still be a complication of the API surface for a benefit we should be able to get another way.

(I was also sorta hoping that we could transition `WindowBuilder` to using `&mut self` at some point, in which case your proposed solution would be harder to work with).

Reply:
> I can do this all myself in a separate patch though.

Yeah, in any case, let's keep this focused on the now added lifetime parameter.

@notgull: Maybe we should wait on changing that for this PR, and just keep the `RawWindowHandle` in there? That'd mean `with_parent_window` is still `unsafe`, but that's just the status quo, so not that bad.

Reply:
I've removed this change for now.